GliaWidgets/Sources/OperatorRequestHandlerService/OperatorRequestHandlerService.swift

```diff
@@ -0,0 +1,309 @@
+        accepted: @escaping () -> Void,
+        declined: @escaping () -> Void,
+        answer: @escaping CoreSdkClient.AnswerWithSuccessBlock
+    ) {
+        logTwoWayVideoRequested()
+
+        let acceptedOffer: () -> Void = { [weak self] in
+            self?.dismiss()
+            self?.logUpgradeOfferAccepted()
+            accepted()
+            answer(true, nil)
+        }
+
+        let declinedOffer: () -> Void = { [weak self] in
+            self?.dismiss()
+            self?.logUpgradeOfferDeclined()
+            declined()
+            answer(false, nil)
+        }
+
+        let alertKind: AlertViewController.Kind = .singleMediaUpgrade(
+            config,
+            accepted: acceptedOffer,
+            declined: declinedOffer
+        )
+
+        let alert = makeAlert(kind: alertKind)
+        present(alert)
+    }
+
+    func makeAudioUpgradeAlert(
+        with config: SingleMediaUpgradeAlertConfiguration,
+        offer: CoreSdkClient.MediaUpgradeOffer,
+        accepted: @escaping () -> Void,
+        declined: @escaping () -> Void,
+        answer: @escaping CoreSdkClient.AnswerWithSuccessBlock
+    ) {
+        logAudioUpgradeRequested()
+
+        let acceptedOffer: () -> Void = { [weak self] in
+            self?.dismiss()
+            self?.logUpgradeOfferAccepted()
+            accepted()
+            answer(true, nil)
+        }
+
+        let declinedOffer: () -> Void = { [weak self] in
+            self?.dismiss()
+            self?.logUpgradeOfferDeclined()
+            declined()
+            answer(false, nil)
+        }
+
+        let alertKind: AlertViewController.Kind = .singleMediaUpgrade(
+            config,
+            accepted: acceptedOffer,
+            declined: declinedOffer
+        )
+
+        let alert = makeAlert(kind: alertKind)
+        present(alert)
+    }
+
+    func dismiss() {
+        alertViewController?.dismiss(animated: true)
+        alertViewController = nil
+        alertWindow?.isHidden = true
+        alertWindow = nil
+    }
+
+    func present(_ alert: AlertViewController) {
+        alertViewController = alert
+        replacePresentedOfferIfPossible(with: alert)
+    }
+
+    func makeAlert(kind: AlertViewController.Kind) -> AlertViewController {
+        .init(kind: kind, viewFactory: viewFactory)
+    }
+
+    func replacePresentedOfferIfPossible(with offer: Replaceable) {
+        guard let alertViewController else {
+            presentAlertViewController(offer)
+            return
+        }
+
+        guard alertViewController.isReplaceable(with: offer) else { return }
+        alertViewController.dismiss(animated: true) { [weak self] in
+            self?.presentAlertViewController(offer)
+        }
+    }
+
+    func alertConfiguration() -> AlertConfiguration {
+        return viewFactory.theme.alertConfiguration
+    }
+}
+
+// MARK: - Logger
+private extension OperatorRequestHandlerService {
+    func logAudioUpgradeRequested() {
```

Comment:
This makes logger tracing context (function name etc) as `logAudioUpgradeRequested` instead of where it is really invoked, which does not bring much value in my opinion. To remedy this you can try to:
a) make all these log functions inlinable, annotating them with `@inline(__always)` and check if they will be inlined in places where it is invoked and capture correct context.
b) explicitly add parameters that capture context (line number, function etc.)  and pass them further to `log.info`).

Reply:
fixed [2bff344](https://github.com/salemove/ios-sdk-widgets/pull/885/commits/2bff344a639d14c0048afecba3007635c01d515e)

Reply:
@rasmustautsglia So these function are inlined and capture context from scopes that invoke them?

Reply:
fixed [c085b79](https://github.com/salemove/ios-sdk-widgets/pull/885/commits/c085b790275c335b280ba63eee44fcbb147e6ab3)